Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs

```diff
@@ -0,0 +1,478 @@
+                }
+                else if (mItem.Consumable.Value > 0)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.ItemDescription.ConsumableTypes[(int)mItem.Consumable.Type], mItem.Consumable.Value.ToString());
+                }
+                else if (mItem.Consumable.Percentage > 0)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.ItemDescription.ConsumableTypes[(int)mItem.Consumable.Type], Strings.ItemDescription.Percentage.ToString(mItem.Consumable.Percentage));
+                }
+            }
+
+            // Resize and position the container.
+            mDetailRows.SizeToChildren(true, true);
+        }
+
+        protected void SetupSpellInfo()
+        {
+            // Add a divider.
+            mDivider3 = AddDivider();
+
+            // Add a row component.
+            mDetailRows = AddRowContainer();
+
+            // Spell data.
+            if (mItem.Spell != null)
+            {
+                if (mItem.QuickCast)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.ItemDescription.CastSpell.ToString(mItem.Spell.Name), string.Empty);
+                }
+                else
+                {
+                    mDetailRows.AddKeyValueRow(Strings.ItemDescription.TeachSpell.ToString(mItem.Spell.Name), string.Empty);
+                }
+
+                if (mItem.SingleUse)
+                {
+                    mDetailRows.AddKeyValueRow(Strings.ItemDescription.SingleUse, string.Empty);
+                }
+            }
+
+            // Resize and position the container.
+            mDetailRows.SizeToChildren(true, true);
+        }
+
+        protected void SetupBagInfo()
+        {
+            // Add a divider.
+            mDivider3 = AddDivider();
+
+            // Add a row component.
+            mDetailRows = AddRowContainer();
+
+            // Bag data.
+            mDetailRows.AddKeyValueRow(Strings.ItemDescription.BagSlots, mItem.SlotCount.ToString());
+
+            // Resize and position the container.
+            mDetailRows.SizeToChildren(true, true);
+        }
+
+        protected void SetupValueInfo()
+        {
+            // Add a divider.
+            mDivider4 = AddDivider();
+
+            // Add a row component.
+            mValueInfo = AddRowContainer();
+
+            // Display shop value.
+            mValueInfo.AddKeyValueRow(mValueLabel, string.Empty);
+
+            // Resize and position the container.
+            mValueInfo.SizeToChildren(true, true);
+        }
+
+        protected void CorrectDisplay()
+        {
+            // Set the correct width.
+            mContainer.SizeToChildren(true, false);
+
+            // Adjust all components.
+            mHeader?.CorrectWidth();
```

Comment:
Maybe we can use a List of components instead?

Reply:
Working on making this more generic and on the DescriptionWindowBase now yes. :)